pkg/vsphere/tasks/waiter.go

```diff
@@ -91,15 +91,19 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 
 func isTaskInProgress(err error) bool {
 	if soap.IsSoapFault(err) {
-		if _, ok := soap.ToSoapFault(err).VimFault().(types.TaskInProgress); ok {
+		f := soap.ToSoapFault(err)
+		if _, ok := f.VimFault().(types.TaskInProgress); ok {
 			return true
 		}
+		log.Debugf("unexpected soap fault on task retry : %s", spew.Sdump(f))
```

Comment:
I introduced spew in this area since we have had problems in the past figuring out exactly what some unexpected faults are. This is more specifically for faults that are hard to reproduce, in those cases I think it would be useful to have as much information about that object as possible. I can change that section to using `%#v`, I plan to leave the others as is if you have no objections. @dougm